templates/default/replication/replication_monitor.sh.erb

```diff
@@ -95,12 +97,37 @@ _channel_takeover() {
     fi
 }
 
+_purge_binlog() {
+    local latest_applied_epoch=$1
+    if [[ "$latest_applied_epoch" == "NULL" ]]; then
+        _log_warn "Latest applied epoch is NULL. Skip purging binlog files"
+        return
+    fi
+
+    # Query MySQL on Primary site which binlog file maps to the last applied epoch
+    binlog=$($MYSQL_BIN_DIR/mysql -h $primary_ip -u $REPLICATION_USER -p$REPLICATION_PASSWORD -Nse "SELECT File FROM mysql.ndb_binlog_index WHERE epoch = $latest_epoch;")
+    if [[ "$binlog" != "NULL" ]]; then
+        _log_info "Purging binary logs up to $binlog"
+        # Purge binlogs up to (exclusive) last applied binlog
+        $MYSQL_BIN_DIR/mysql -h $primary_ip -u $REPLICATION_USER -p$REPLICATION_PASSWORD -Nse "PURGE BINARY LOGS TO '$binlog'"
+
+        _log_info "Deleted applied binary logs"
+        _log_info "Deleting entries from mysql.ndb_binlog_index where epoch < $latest_applied_epoch"
+        # we also need to delete records from ndb_binlog_index periodically otherwise this table will keep increasing
+        $MYSQL_BIN_DIR/mysql -h $primary_ip -u $REPLICATION_USER -p$REPLICATION_PASSWORD -Nse "DELETE FROM mysql.ndb_binlog_index WHERE epoch < $latest_applied_epoch LIMIT 500;"
```

Comment:
is 500 per 30 seconds enough in all cases to remove the epochs, if you generate 100 per second one will generate 3000 rows per 30 seconds. So in that case this method will not keep up.

Reply:
@mronstro I addressed your comment. The load is sort of variable so we might need to tune these parameters in the future